OpenRA.Platforms.Default/ThreadedGraphicsContext.cs

```diff
@@ -426,9 +441,14 @@ public ITexture CreateTexture()
 			return Send(getCreateTexture);
 		}
 
-		public IVertexBuffer<T> CreateVertexBuffer<T>(int size) where T : struct
+		public IVertexBuffer<T> CreateEmptyVertexBuffer<T>(int size) where T : struct
+		{
+			return (IVertexBuffer<T>)Send(getCreateEmptyVertexBuffer, (size, typeof(T)));
+		}
+
+		public IVertexBuffer<T> CreateVertexBuffer<T>(ref T[] data, bool dynamic = true) where T : struct
 		{
-			return (IVertexBuffer<T>)Send(getCreateVertexBuffer, (size, typeof(T)));
+			return (IVertexBuffer<T>)Send(getCreateVertexBuffer, ((object)data, dynamic, typeof(T)));
```

Comment:
We declare `ref T[] data` - but never actually change out the `ref`!

Simplify the callers and remove the `ref` requirement.

(The alternative is that we allocate a temp array and reassign so we can post like `ThreadedVertexBuffer<T>.SetData(ref T[] vertices, int length)` does, but creating vertex buffers is rare so I don't think we need to bother with that and can afford to block instead)

Reply:
The threaded queue messes that structure up. It would have worked in a non-threaded context